### PR TITLE
[spi_passthru] Test the jedec_id

### PR DESCRIPTION
**Pull request by cfrantz**
1. Create the basic framework for the spi_passthru test.
2. Test the JEDEC ID feature of thee spi_device.

### CODE REVIEW

File: sw/device/tests/spi_passthru_test.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG(.enable_uart_flow_control = true);
+
+static dif_spi_device_handle_t spid;
+
+status_t configure_jedec_id(ujson_t *uj, dif_spi_device_handle_t *spid) {
+  config_jedec_id_t config;
+  TRY(ujson_deserialize_config_jedec_id_t(uj, &config));
+  dif_spi_device_flash_id_t id = {
```
Comment (a-will) [inline sw/device/tests/spi_passthru_test.c:29]: Just a thought: The logic for setting up the command and IDs could probably go into spi_device_testutils at some point.

File: sw/host/tests/chip/spi_passthru/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    log::info!("jedec_id = {:x?}", jedec_id);
+    assert_eq!(jedec_id, [
+        // 3 continuation codes of 0x7F.
+        0x7f, 0x7f, 0x7f,
+        // Manufacturer ID of 0x56.
+        0x56,
+        // Device ID 0x1234 (in little endian order).
+        0x34, 0x12,
+        // All zeros up to read length.
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
```
Comment (a-will) [inline sw/host/tests/chip/spi_passthru/src/main.rs:58]: At some point, we might want to loosen this particular restriction and only take the slice that is valid data. I'm not sure if SPI flash chips are required to return any particular value after reading beyond the ID bytes.

**Review by a-will — APPROVED** [review]
LGTM

File: sw/device/tests/spi_passthru_test.c (hunk 1 above)
Comment (cfrantz) [reply sw/device/tests/spi_passthru_test.c:29]: I think we can make a judgement about how to handle that in the follow-on PRs.

File: sw/host/tests/chip/spi_passthru/src/main.rs (hunk 2 above)
Comment (cfrantz) [reply sw/host/tests/chip/spi_passthru/src/main.rs]: Added a comment explaining that the zeros are an OT specific behavior.

**Pull request merged**